### PR TITLE
PP-5170: only process capture on feature flag

### PR DESCRIPTION
**Pull request by sfount**
* introduce lifecycle hook for `QueueMessageReceiver`
* check feature flag for actually processing messages

### CODE REVIEW

**Comment by sfount** [comment]
Waiting on https://github.com/alphagov/pay-connector/pull/1325 -- rebasing to this will allow CI to pass.

**Review by heathd — DISMISSED** [review]
Looks good, thanks!

File: src/main/java/uk/gov/pay/connector/app/ConnectorConfiguration.java [hunk 1]
```diff
@@ -86,6 +86,11 @@
     @JsonProperty("sqsConfig")
     private SqsConfig sqsConfig;
 
+    @Valid
+    @NotNull
+    @JsonProperty("backgroundProcessingEnabled")
+    private Boolean backgroundProcessingEnabled;
```
Comment (kbottla) [inline src/main/java/uk/gov/pay/connector/app/ConnectorConfiguration.java:92]: Can we have it in `CaptureProcessConfig` so capture related config can be at one place ?

Also add to README.md about the `BACKGROUND_PROCESSING_ENABLED`

**Review by kbottla — COMMENTED** [review]
Looks good. Just a minor comment

File: src/main/java/uk/gov/pay/connector/app/ConnectorConfiguration.java (hunk 1 above)
Comment (jankowiakmaria) [reply src/main/java/uk/gov/pay/connector/app/ConnectorConfiguration.java:92]: Can I do it in separate PR or does it need to be a part of this one?

Comment (kbottla) [reply src/main/java/uk/gov/pay/connector/app/ConnectorConfiguration.java:92]: Don't mind if it is refactor as separate PR

**Review by kbottla — APPROVED** [review]

**Pull request merged**